src/pudl/transform/ferc1.py

```diff
@@ -853,6 +964,228 @@ def strip_col_of_non_ints(self, col: pd.Series) -> pd.DataFrame:
+            dropped. If both input dataframes are empty, an empty dataframe is returned.
+        """
+        drop_cols = ["filing_name", "index"]
+        # Ignore errors in case not all drop_cols are present.
+        instant = raw_xbrl_instant.drop(columns=drop_cols, errors="ignore")
+        duration = raw_xbrl_duration.drop(columns=drop_cols, errors="ignore")
+
+        if instant.empty:
+            logger.debug(
+                f"{self.table_id.value}: No XBRL instant table found, returning the "
+                "duration table."
+            )
+            return duration
+        if duration.empty:
+            logger.debug(
+                f"{self.table_id.value}: No XBRL duration table found, returning "
+                "instant table."
+            )
+            return instant
+
+        instant_axes = [
+            col for col in raw_xbrl_instant.columns if col.endswith("_axis")
+        ]
+        duration_axes = [
+            col for col in raw_xbrl_duration.columns if col.endswith("_axis")
+        ]
+        if set(instant_axes) != set(duration_axes):
+            raise ValueError(
+                f"{self.table_id.value}: Instant and Duration XBRL Axes do not match.\n"
+                f"    instant: {instant_axes}\n"
+                f"    duration: {duration_axes}"
+            )
+
+        # Everything above here is the same as the parent function, but here we need
+        # to reshape the instant table such that end-of-last-year becomes
+        # beginning-of-this-year in a separate column. Probably going to need to factor
+        # out the shared vs. different code into separate methods once we know what this
+        # should look like.
+        instant = self.reshape_starting_ending_balances(instant)
+        return pd.concat(
+            [
+                instant.set_index(["report_year", "entity_id"] + instant_axes),
+                duration.set_index(["report_year", "entity_id"] + duration_axes),
+            ],
+            axis="columns",
+        ).reset_index()
+
+    @cache_df("reshape_instant")
+    def reshape_starting_ending_balances(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Convert row-oriented end-of-year balances to starting/ending balance cols."""
+        df["year"] = pd.to_datetime(df["date"]).dt.year
+        df.loc[df.report_year == (df.year + 1), "balance_type"] = "starting_balance"
+        df.loc[df.report_year == df.year, "balance_type"] = "ending_balance"
+        # If there are other years something is wrong.
+        assert df.balance_type.notna().all()
+        df = (
+            df.drop(["year", "date"], axis="columns")
+            .set_index(["entity_id", "report_year", "balance_type"])
+            .unstack()
+        )
+        df.columns = ["_".join(items) for items in df.columns.to_flat_index()]
+        return df.reset_index()
+
+    def tidy_ferc_accounts_xbrl(self, df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
I know the `MultiIndex` manipulation in this method is gawdawful and could probably be much simpler and more concise and easier to parametrize such that it can work in many similar situations, but I don't know what the right approach is. If you have suggestions @TrentonBush I would love to hear them.

Ideally we'd be able to use the set of column suffixes (which become the final output columns) as the main parameter, and have it construct an appropriate multi-index based on the column labels, and then stack to turn the non-suffix portions of the column names into a categorical column (here we're going from almost 500 columns to 6 columns).